transmit-lib/src/cmr/transmit/http_helper.clj

```diff
@@ -45,6 +45,22 @@
              body)
      :content-type content-type}))
 
+(defn- http-response->raw-response-with-headers
+  "Parses a clj-http response and returns only the keys that we would normally be interested in.
+  Parses the json in the body if the content type is JSON. The \"raw\" response body is considered
+  useful in cases where the exact status code is required such as in testing."
+  [{:keys [status body headers]}]
+  (let [content-type (mt/mime-type->format
+                       (mt/content-type-mime-type headers)
+                       ;; don't return a default
+                       nil)]
+    {:status status
+     :body (if (= content-type :json)
+             (safe-parse-json body)
+             body)
+     :content-type content-type
+     :headers headers}))
```

Comment:
Why do we need this function?  Can't we just add the headers to http-response->raw-response function on line 33?

Reply:
I did that first. and it causes a lot of test failures.